server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -54,52 +52,45 @@
 
 import java.io.IOException;
 import java.time.ZoneId;
-import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
 /**
  * This defines the core properties and functions to operate on a field.
  */
-public abstract class MappedFieldType extends FieldType {
+public abstract class MappedFieldType {
 
-    private String name;
+    private final String name;
+    private final boolean docValues;
+    private final boolean isSearchable;
     private float boost;
-    // TODO: remove this docvalues flag and use docValuesType
-    private boolean docValues;
     private NamedAnalyzer indexAnalyzer;
     private NamedAnalyzer searchAnalyzer;
     private NamedAnalyzer searchQuoteAnalyzer;
     private SimilarityProvider similarity;
-    private Object nullValue;
```

Comment:
nullValue and nullValueAsString are now dealt with directly by FieldMappers that require it